linera-chain/src/manager.rs

```diff
@@ -685,7 +691,7 @@ where
     /// Sets the signed proposal, if it is newer than the known one, and not from a single-leader
     /// round. Returns whether it was updated.
     pub fn update_signed_proposal(&mut self, proposal: &BlockProposal) -> bool {
-        if proposal.content.round > Round::MultiLeader(u32::MAX) {
+        if proposal.content.round > Round::SingleLeader(0) {
```

Comment:
The comment says that the signed proposal canot be from the single-leader round. Can you explain why it can be now?